app/harvesters/hathi_harvester.rb

```diff
@@ -0,0 +1,104 @@
+require 'net/http'
+require 'rubygems/package'
+require 'zlib'
+
+
+class HathiHarvester < Krikri::Harvesters::MarcXMLHarvester
+  BASE_URL = 'http://mirlyn-aleph.lib.umich.edu/mdp/dpla_metadata/'
+
+  def initialize(opts = {})
+    if !(opts[:url] || opts[:dump_file])
+      raise "You must provide either `:url` or `:dump_file`"
+    end
+
+    opts[:uri] = (opts[:url] || opts[:dump_file])
```

Comment:
I see what's going on here, but it feels a little bit hacky.

The thinking behind making `:uri` a [required option](https://github.com/dpla/KriKri/blob/develop/lib/krikri/harvester.rb#L73) for all harvesters was that there would be some consistency in what get's passed in. E.g. in a form to create a harvester, you might have a first step where you always take `:uri`.  We should either drop that idea, and make `:uri` optional in `Harvester#initialize` or simply take both `:url` and `:dump_file` at `:uri` and use `File.exists?(uri)` or similar to determine the behavior.